website/docs/configuration.mdx

```diff
@@ -6,10 +6,87 @@ description: Configuring your site's behavior through docusaurus.config.js and m
 
 Keeping a well-maintained `docusaurus.config.js` helps you, your collaborators, and your open source contributors to be able to focus on documentation while still being able to customize the site.
 
+## Syntax to declare docusaurus.config.js
+
+The init template `docusaurus.config.js` uses the `const config = { ... }` and it is equivalent to `module.exports = { ... }` shown throughout the examples.
+There are various equivalent ways to declare your Docusaurus config, such as using creator functions and async code.
+See: [**`docusaurus.config.js` API reference**](api/docusaurus.config.js.mdx)
+
+Here is a verbose example using most of the syntax that is available ([docusaurus.config.js API](api/docusaurus.config.js.mdx), JS, Node, constants, variables, etc.):
+
+```js title="docusaurus.config.js"
+const path = require('path');
+const npm2yarn = require('@docusaurus/remark-plugin-npm2yarn');
+const versions = require('./versions.json');
+const VersionsArchived = require('./versionsArchived.json');
+
+module.exports = async function createConfigAsync() {
+  return {
+    title: 'Docusaurus',
+    tagline: getLocalizedConfigValue('tagline'),
+    organizationName: 'facebook',
+    projectName: 'docusaurus',
+    baseUrl,
+    baseUrlIssueBanner: true,
+    url: 'https://docusaurus.io',
+    themes: ['live-codeblock', ...dogfoodingThemeInstances],
+    plugins: [
+      [
+        require.resolve('./src/plugins/changelog/index.js'),
+        {
+          blogTitle: 'Docusaurus changelog',
+          blogDescription:
+            'Keep yourself up-to-date about new features in every release',
+        },
+      ],
+      ['content-docs',
+        /** @type {import('@docusaurus/plugin-content-docs').Options} */
+        ({
+          id: 'community',
+          path: 'community',
+          routeBasePath: 'community',
+          editUrl: ({locale, versionDocsDirPath, docPath}) => {
+            if (locale !== defaultLocale) {
+              return `https://crowdin.com/project/docusaurus-v2/${locale}`;
+            }
+            return `https://github.com/facebook/docusaurus/edit/main/website/${versionDocsDirPath}/${docPath}`;
+          },
+          remarkPlugins: [npm2yarn],
+          editCurrentVersion: true,
+          sidebarPath: require.resolve('./sidebarsCommunity.js'),
+          showLastUpdateAuthor: true,
+          showLastUpdateTime: true,
+        }),
+
+      ],
+    ],
+  };
+};
+
+```
+
+Or using a Promise:
+```js title="docusaurus.config.js"
+module.exports = Promise.resolve({
+  // site config...
+});
+
+```
+
+Since a config file is Javascript then import/require can be utilized to pull in libraries to help.  (For advanced users, just know that we )
+
+// TODO: link for nodejs import/require.  And answer question about interoperability for CommonJS and ECMAScript modules.
```

Comment:
That's too verbose

What I want is something concice with many small examples like this

<img width="990" alt="CleanShot 2023-05-04 at 11 21 55@2x" src="https://user-images.githubusercontent.com/749374/236163913-d134091d-2c54-4dfe-942c-ca89cbe40c12.png">